tracee-ebpf/tracee/events_processor.go

```diff
@@ -212,7 +212,11 @@ func (t *Tracee) processEvent(ctx *context, args map[string]interface{}, argMeta
 		if !ok {
 			return fmt.Errorf("error parsing cgroup_mkdir args")
 		}
-		t.containers.CgroupUpdate(cgroupId, path)
+		info, err := t.containers.CgroupUpdate(cgroupId, path)
+		if err == nil && info.ContainerId == "" {
+			// If not a new container (no regex match) - remove from the bpf container_map
+			t.containers.RemoveFromBpfMap(t.bpfModule, cgroupId)
```

Comment:
Okay, so.. just to check if I understood, you remove it from the container_map but you keep it in the t.containers with empty CgroupInfo{} so it can be removed whenever a cgroup rmdir event happens. Is that it ?

Reply:
Yes.
Only the cgroup_mkdir and cgroup_rmdir events update the USERSPACE containers map (opposed to the bpf containers map that only keeps track of "real" containers). We keep "non-containers" cgroups so it will be found by CgroupExists() in processEvents (and will give an empty container id)